gadget/ondisk.go

```diff
@@ -363,6 +363,23 @@ func BuildPartitionList(dl *OnDiskVolume, pv *LaidOutVolume) (sfdiskInput *bytes
 	return buf, toBeCreated
 }
 
+// UpdatePartitionList re-reads the partitioning data from the device and
+// updates the partition list in the specified volume.
+func UpdatePartitionList(dl *OnDiskVolume) error {
```

Comment:
(cosmetic wondering) I wonder if this should be a helper on "dl *OnDiskVolume". But I don't know the code well enough to feel confident that this fit the style (or does not).

Something like:
```
diff --git a/gadget/install/partition.go b/gadget/install/partition.go
index 27c9e6bf44..bfbd2f8ef8 100644
--- a/gadget/install/partition.go
+++ b/gadget/install/partition.go
@@ -92,7 +92,7 @@ func removeCreatedPartitions(dl *gadget.OnDiskVolume) error {
        }
 
        // Re-read the partition table from the device to update our partition list
-       if err := gadget.UpdatePartitionList(dl); err != nil {
+       if err := dl.UpdatePartitionList(); err != nil {
                return err
        }
 
diff --git a/gadget/ondisk.go b/gadget/ondisk.go
index 35fcc6918f..aaefd4bbd6 100644
--- a/gadget/ondisk.go
+++ b/gadget/ondisk.go
@@ -365,7 +365,7 @@ func BuildPartitionList(dl *OnDiskVolume, pv *LaidOutVolume) (sfdiskInput *bytes
 
 // UpdatePartitionList re-reads the partitioning data from the device and
 // updates the partition list in the specified volume.
-func UpdatePartitionList(dl *OnDiskVolume) error {
+func (dl *OnDiskVolume) UpdatePartitionList() error {
        layout, err := OnDiskVolumeFromDevice(dl.Device)
        if err != nil {
                return fmt.Errorf("cannot read disk layout: %v", err)
diff --git a/gadget/ondisk_test.go b/gadget/ondisk_test.go
index 28db2c25e8..ec854b0452 100644
--- a/gadget/ondisk_test.go
+++ b/gadget/ondisk_test.go
@@ -478,7 +478,7 @@ exit 0`
        defer cmdLsblk.Restore()
 
        // update the partition list
-       err = gadget.UpdatePartitionList(dl)
+       err = dl.UpdatePartitionList()
        c.Assert(err, IsNil)
 
        // check if the partition list was updated
```

Reply:
The reason it was declared as a separate function is to be in line with `BuildPartitionList()` and `CreatedDuringInstall()` which are not methods on OnDiskVolume, but having it as you suggested also makes sense. A major refactoring should take place at some point to merge OnDiskVolume/Structure and LaidOutVolume/Structure, and we can review these calls when this happens.